ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -419,65 +367,68 @@ func resourceIBMCbrRuleRead(context context.Context, d *schema.ResourceData, met
 func resourceIBMCbrRuleUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
 	if err != nil {
-		return diag.FromErr(err)
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_cbr_rule", "update", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
 	}
 
 	replaceRuleOptions := &contextbasedrestrictionsv1.ReplaceRuleOptions{}
 
 	replaceRuleOptions.SetRuleID(d.Id())
+	if _, ok := d.GetOk("x_correlation_id"); ok {
+		replaceRuleOptions.SetXCorrelationID(d.Get("x_correlation_id").(string))
+	}
+	if _, ok := d.GetOk("transaction_id"); ok {
+		replaceRuleOptions.SetTransactionID(d.Get("transaction_id").(string))
+	}
 	if _, ok := d.GetOk("description"); ok {
 		replaceRuleOptions.SetDescription(d.Get("description").(string))
 	}
-	contexts := []contextbasedrestrictionsv1.RuleContext{}
 	if _, ok := d.GetOk("contexts"); ok {
-		for _, e := range d.Get("contexts").([]interface{}) {
-			value := e.(map[string]interface{})
-			contextsItem, err := resourceIBMCbrRuleMapToRuleContext(value)
+		var contexts []contextbasedrestrictionsv1.RuleContext
+		for _, v := range d.Get("contexts").([]interface{}) {
+			value := v.(map[string]interface{})
+			contextsItem, err := ResourceIBMCbrRuleMapToRuleContext(value)
 			if err != nil {
-				return diag.FromErr(err)
+				return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_cbr_rule", "update", "ResourceIBMCbrRuleMapToRuleContext").GetDiag()
 			}
 			contexts = append(contexts, *contextsItem)
 		}
+		replaceRuleOptions.SetContexts(contexts)
```

Comment:
What is the reason for this change? The API won't accept a zone create with missing contexts. The previous code ensured that it is present (and empty) if no contexts are specified.

Reply:
changed it to previous code